kafka/src/main/java/io/specmesh/kafka/Provisioner.java

```diff
@@ -127,4 +133,82 @@ static ParsedSchema getSchema(final String topicName, final String schemaRef, fi
+     * @throws TimeoutException
+     *             stalled
+     */
+    public static void provision(final KafkaApiSpec apiSpec, final String schemaResources,
+            final AdminClient adminClient, final SchemaRegistryClient schemaRegistryClient)
+            throws ExecutionException, InterruptedException, TimeoutException {
+        provisionTopics(apiSpec, adminClient);
+        provisionSchemas(apiSpec, schemaResources, schemaRegistryClient);
+        provisionAcls(apiSpec, adminClient);
+
+    }
+
+    /**
+     * setup sasl_plain auth creds
+     *
+     * @param principle
+     *            user name
+     * @param secret
+     *            secret
+     * @return client creds map
+     */
+    public static Map<String, Object> clientAuthProperties(final String principle, final String secret) {
+        return Map.of("sasl.mechanism", "PLAIN", AdminClientConfig.SECURITY_PROTOCOL_CONFIG, "SASL_PLAINTEXT",
+                "sasl.jaas.config", String.format("org.apache.kafka.common.security.plain.PlainLoginModule required "
+                        + "username=\"%s\" password=\"%s\";", principle, secret));
+
+    }
+
+    /**
+     * Configure environment for SASL_PLAIN auth with 2 sets of users
+     *
+     * @param domainUser
+     *            owner user
+     * @param domainSecret
+     *            their secret
+     * @param otherDomainUser
+     *            other user
+     * @param otherDomainSecret
+     *            their secret
+     * @return env map for broker
+     */
+    public static Map<String, String> testAuthorizerConfig(final String domainUser, final String domainSecret,
+            final String otherDomainUser, final String otherDomainSecret) {
+        final Map<String, String> env = new LinkedHashMap<>();
+        env.put("KAFKA_AUTO_CREATE_TOPICS_ENABLE", "false");
+        env.put("KAFKA_ALLOW_EVERYONE_IF_NO_ACL_FOUND", "true");
+        env.put("KAFKA_AUTHORIZER_CLASS_NAME", "kafka.security.authorizer.AclAuthorizer");
+        env.put("KAFKA_SUPER_USERS", "User:OnlySuperUser");
+        env.put("KAFKA_SASL_ENABLED_MECHANISMS", "PLAIN,SASL_PLAINTEXT");
+
+        env.put("KAFKA_LISTENER_SECURITY_PROTOCOL_MAP", "BROKER:PLAINTEXT,PLAINTEXT:SASL_PLAINTEXT");
+        env.put("KAFKA_LISTENER_NAME_PLAINTEXT_SASL_ENABLED_MECHANISMS", "PLAIN");
+        env.put("KAFKA_LISTENER_NAME_PLAINTEXT_PLAIN_SASL_JAAS_CONFIG",
+                "org.apache.kafka.common.security.plain.PlainLoginModule required "
+                        + "username=\"admin\" password=\"admin-secret\" user_admin=\"admin-secret\" "
+                        + String.format("user_%s=\"%s\" ", domainUser, domainSecret /* secret */)
+                        + String.format("user_%s=\"%s\";", otherDomainUser, otherDomainSecret) /* secret */);
+
+        env.put("KAFKA_SASL_JAAS_CONFIG", "org.apache.kafka.common.security.plain.PlainLoginModule required "
+                + "username=\"admin\" " + "password=\"admin-secret\";");
```

Comment:
This looks way more complicated than I'd expect / remember.  e.g. having two sets of JAAS config.  

Would be good to minimise this config. i.e. trim it down to the minimum required to run. However, I guess we can address when we move beyond the POC.

Reply:
simplest I can make it... will more look later